lib/engine/game/g_1822/step/route.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/route'
+
+module Engine
+  module Game
+    module G1822
+      module Step
+        class Route < Engine::Step::Route
+          def actions(entity)
+            return [] if !entity.operator? || @game.route_trains(entity).empty? || !@game.can_run_route?(entity)
+
+            @pullman_train ||= nil
+            actions = ACTIONS.dup
+            actions << 'choose' if !@pullman_train && find_pullman_train(entity)
+            actions
+          end
+
+          def attach_pullman
+            @orginal_train = @pullman_train.dup
```

Comment:
Unless the detach has to be done immediately this won't work as the route step will be different in the next OR and member variables will be cleared.

Also this probably needs to take into account the case of train discards and ensure the Pullman gets removed.

Reply:
Yes i know, you have to attach the pullman the next or. But this how it should be done since you have the option to attach it to another train.

Reply:
The pullman detaches before you leave the route. So it should be fine.

Reply:
Yep sorry didn't read closely enough